packages/client/hmi-client/src/services/dataset.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Datasets
+ */
+
+import API from '@/api/api';
+import { Dataset } from '@/types/Dataset';
+
+/**
+ * Get all datasets
+ * @return Array<Dataset>|null - the list of all datasets, or null if none returned by API
+ */
+async function getAll(): Promise<Dataset[] | null> {
+	const response = await API.get('/datasets');
+	return response?.data ?? null;
+}
+
+const getDataset = async (datasetId: string) => API.get(`/datasets/${datasetId}`);
```

Comment:
Why this function is not defined like `getAll`. Then in the `Dataset.vue` you need to handle null and `response.data`?

Reply:
The code is copied from the example in model.ts, but I have refactored this one as suggested